07/hangman/hangman.py

```diff
@@ -0,0 +1,35 @@
+
+from string import ascii_lowercase
+
+
+def change(game_field, game_random_word):
+    '''Vymeni pismeno za '-' '''
+
+    while True:      # Overeni vstupu
+        letter = input('Zadej pismeno: ').lower()
+        if letter in ascii_lowercase and len(letter) == 1:      # Jestlize neni pismeno v abecede
+            break
+        print('Toto neni jedno pismeno!')
+
+    correct_letter = letter in game_random_word      # Pismeno je v hledanem slove
+
+    if correct_letter:
+        position = -1  
+        for i in game_random_word:       # Vypocet pozice pismene v hracim slove
+            position += 1
+            if i == letter:
+                break
```

Comment:
Na tomhle místě už víš, že je pravda, že `letter in game_random_word`. Takže místo těchto 4 řádků bys mohla použít buď [metodu `str.index()`](https://docs.python.org/3/library/stdtypes.html#str.index), která ti vrátí index prvního výskytu substringu ve stringu _(a nebo vyhodí `ValueError`, pokud se tam nenajde - což se ale v tomhle případě stát nemůže, protože jsi právě už přes `... in ...` zkontrolovala, že to tam fakt je 😉 - ale kdyby jsi to nejdřív nekontrolovala, tak právě ideální kandidát na `try...except` blok kolem toho volání metody `str.index()`)_

```python
position = game_random_word.index(letter)
```

```python
"ahoj".index("a")  # 0
"ahoj".index("j")  # 3
"ahoj".index("hoj")  # 1
"ahoj".index("X")  # ValueError
```

No a nebo _(pokud bys nejdřív `letter in game_random_word` třeba nekontrolovala)_ by se ještě dala použít [metoda `str.find()`](https://docs.python.org/3/library/stdtypes.html#str.find), která dělá to samé, ale místo vyhození `exception` ti - pokud tam ten substring neexistuje - vrátí `-1`.

```python
"ahoj".find("a")  # 0
"ahoj".find("j")  # 3
"ahoj".find("hoj")  # 1
"ahoj".find("X")  # -1
```